plugins/newsdownloader.koplugin/main.lua

```diff
@@ -153,12 +158,40 @@ function NewsDownloader:processFeedSource(url, limit)
     http.request({ url = url, sink = ltn12.sink.table(resp_lines), })
     local feeds = deserializeXMLString(table.concat(resp_lines))
     if not feeds then return end
-    if not feeds.rss or not feeds.rss.channel
-            or not feeds.rss.channel.title or not feeds.rss.channel.item then
+    
+    local is_valid_rss = feeds.rss and feeds.rss.channel and feeds.rss.channel.title and feeds.rss.channel.item;
+    local is_valid_atom = feeds.feed and feeds.feed.title and feeds.feed.entry.title and feeds.feed.entry.link;
```

Comment:
It was just to it has structure code's expecting, and so shouldn't cause any problems during further processing.